pytest_html/plugin.py

```diff
@@ -215,23 +220,55 @@ def pytest_sessionfinish(self, session):
             html.title('Test Report'),
             html.style(raw(style_css)))
 
+        class Outcome:
+
+            def __init__(self, outcome, total=0, label=None,
+                         test_result=None, class_html=None):
+                self.outcome = outcome
+                self.label = label or outcome
+                self.class_html = class_html or outcome
+                self.total = total
+                self.test_result = test_result or outcome
+
+                self.generate_checkbox()
+                self.generate_summary_item()
+
+            def generate_checkbox(self):
+                data_test_result = {"data-test-result":
+                                    self.test_result.lower()}
+                if(self.total > 0):
+                    self.checkbox = html.input(type='checkbox',
+                                               checked='true',
+                                               onChange='filter_table(this)',
+                                               name='filter_checkbox',
+                                               **data_test_result)
+                else:
+                    self.checkbox = html.input(type='checkbox',
+                                               disabled='true')
+
+            def generate_summary_item(self):
+                self.summary_item = html.span('{0} {1}'.
+                                              format(self.total, self.label),
+                                              class_=self.class_html)
+
+        outcomes = [Outcome('passed', self.passed),
+                    Outcome('skipped', self.skipped),
+                    Outcome('failed', self.failed),
+                    Outcome('error', self.errors, 'errors'),
+                    Outcome('xfailed', self.xfailed,
+                            'expected failures', 'xfailed', 'skipped'),
```

Comment:
The `test_result` argument is not needed as it matches the outcome. You can avoid specifying it by passing the keywords for the optional arguments: `Outcome('xfailed', self.xfailed, label='expected failures', class_html='skipped'`